zio-http/src/main/scala/zio/http/HandlerConstructor.scala

```diff
@@ -0,0 +1,538 @@
+    }
+
+  implicit def function5ResponseIsHandlerConstructor[In1, In2, In3, In4, In5]: HandlerConstructor.Typed[
+    (In1, In2, In3, In4, In5) => Response,
+    Any,
+    Nothing,
+    (In1, In2, In3, In4, In5),
+    Response,
+  ] =
+    new HandlerConstructor[(In1, In2, In3, In4, In5) => Response] {
+      type Env = Any
+      type Err = Nothing
+      type In  = (In1, In2, In3, In4, In5)
+      type Out = Response
+      type Z   = (In1, In2, In3, In4, In5) => Response
+
+      def toHandler(z: => Z): Handler[Env, Err, In, Out] =
+        Handler.fromFunction { case (in1, in2, in3, in4, in5) => z(in1, in2, in3, in4, in5) }
+    }
+
+  implicit def function6ResponseIsHandlerConstructor[In1, In2, In3, In4, In5, In6]: HandlerConstructor.Typed[
+    (In1, In2, In3, In4, In5, In6) => Response,
+    Any,
+    Nothing,
+    (In1, In2, In3, In4, In5, In6),
+    Response,
+  ] =
+    new HandlerConstructor[(In1, In2, In3, In4, In5, In6) => Response] {
+      type Env = Any
+      type Err = Nothing
+      type In  = (In1, In2, In3, In4, In5, In6)
+      type Out = Response
+      type Z   = (In1, In2, In3, In4, In5, In6) => Response
+
+      def toHandler(z: => Z): Handler[Env, Err, In, Out] =
+        Handler.fromFunction { case (in1, in2, in3, in4, in5, in6) => z(in1, in2, in3, in4, in5, in6) }
+    }
+
+  implicit def function7ResponseIsHandlerConstructor[In1, In2, In3, In4, In5, In6, In7]: HandlerConstructor.Typed[
+    (In1, In2, In3, In4, In5, In6, In7) => Response,
+    Any,
+    Nothing,
+    (In1, In2, In3, In4, In5, In6, In7),
+    Response,
+  ] =
+    new HandlerConstructor[(In1, In2, In3, In4, In5, In6, In7) => Response] {
+      type Env = Any
+      type Err = Nothing
+      type In  = (In1, In2, In3, In4, In5, In6, In7)
+      type Out = Response
+      type Z   = (In1, In2, In3, In4, In5, In6, In7) => Response
+
+      def toHandler(z: => Z): Handler[Env, Err, In, Out] =
+        Handler.fromFunction { case (in1, in2, in3, in4, in5, in6, in7) => z(in1, in2, in3, in4, in5, in6, in7) }
+    }
+
+  implicit def function8ResponseIsHandlerConstructor[In1, In2, In3, In4, In5, In6, In7, In8]: HandlerConstructor.Typed[
+    (In1, In2, In3, In4, In5, In6, In7, In8) => Response,
+    Any,
+    Nothing,
+    (In1, In2, In3, In4, In5, In6, In7, In8),
+    Response,
+  ] =
+    new HandlerConstructor[(In1, In2, In3, In4, In5, In6, In7, In8) => Response] {
+      type Env = Any
+      type Err = Nothing
+      type In  = (In1, In2, In3, In4, In5, In6, In7, In8)
+      type Out = Response
+      type Z   = (In1, In2, In3, In4, In5, In6, In7, In8) => Response
+
+      def toHandler(z: => Z): Handler[Env, Err, In, Out] =
+        Handler.fromFunction { case (in1, in2, in3, in4, in5, in6, in7, in8) =>
+          z(in1, in2, in3, in4, in5, in6, in7, in8)
+        }
+    }
+}
+
+private[http] trait HandlerConstructorLowPriorityImplicits2 extends HandlerConstructorLowPriorityImplicits3 {
```

Comment:
We should make all the variants defined up to the same number of parameters

Reply:
Done, up to 7 for now.